### PR TITLE
[feat]: [PL-19821]: csm ng connector

### PR DESCRIPTION
**Pull request by shreyas-harness**
## Describe your changes
Custom Secret Manager Connector - DTO's , Entity, Mappers , configs and their mappers in registry.
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/36571)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 7 Issues 
* Code issue 7 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/36571)

File: 890-sm-core/src/main/java/io/harness/ng/CustomSecretManagerConfigDTOMapper.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import io.harness.secretmanagerclient.dto.CustomSecretManagerConfigDTO.CustomSecretManagerConfigDTOBuilder;
+import io.harness.security.encryption.EncryptionType;
+
+public class CustomSecretManagerConfigDTOMapper {
+  public static CustomSecretManagerConfigDTO getCustomSecretManagerConfigDTO(
+      String accountIdentifier, ConnectorDTO connectorRequestDTO, CustomSecretManagerConnectorDTO connectorDTO) {
+    ConnectorInfoDTO connector = connectorRequestDTO.getConnectorInfo();
+    CustomSecretManagerConfigDTOBuilder<?, ?> builder =
+        CustomSecretManagerConfigDTO.builder()
+            .isDefault(false)
```
Comment (meenakshi-harness) [inline 890-sm-core/src/main/java/io/harness/ng/CustomSecretManagerConfigDTOMapper.java:24]: Why is this set as false?

File: 890-sm-core/src/main/java/io/harness/ng/CustomSecretManagerConfigDTOMapper.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+            .delegateSelectors(connectorDTO.getDelegateSelectors())
+            .onDelegate(connectorDTO.getExecuteOnDelegate())
+            .connectorRef(SecretRefHelper.getSecretConfigString(connectorDTO.getConnectorRef()))
+            .host(connectorDTO.getHost())
+            .workingDirectory(connectorDTO.getWorkingDirectory())
+            .template(connectorDTO.getTemplate());
+
+    if (null != connectorDTO.getConnectorRef() && null != connectorDTO.getConnectorRef().getDecryptedValue()) {
+      builder.connectorRef(String.valueOf(connectorDTO.getConnectorRef().getDecryptedValue()));
```
Comment (meenakshi-harness) [inline 890-sm-core/src/main/java/io/harness/ng/CustomSecretManagerConfigDTOMapper.java:44]: Why we setting connectorRef  with value?

File: 940-secret-manager-client/src/main/java/io/harness/secretmanagerclient/dto/CustomSecretManagerConfigDTO.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+@EqualsAndHashCode(callSuper = true)
+@SuperBuilder
+public class CustomSecretManagerConfigDTO extends SecretManagerConfigDTO implements DelegateSelectable {
+  Set<String> delegateSelectors;
+  private boolean onDelegate;
+
+  @Schema(description = SecretManagerDescriptionConstants.CUSTOM_AUTH_TOKEN) private String connectorRef;
+  private String host;
+  private String workingDirectory;
+  private TemplateLinkConfig template;
```
Comment (meenakshi-harness) [inline 940-secret-manager-client/src/main/java/io/harness/secretmanagerclient/dto/CustomSecretManagerConfigDTO.java:40]: this is not null

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/customsecretmanager/CustomSecretManagerConnectorDTO.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+@AllArgsConstructor
+@ToString(exclude = {"connectorRef"})
+@EqualsAndHashCode(callSuper = true)
+@JsonIgnoreProperties(ignoreUnknown = true)
+@FieldDefaults(level = AccessLevel.PRIVATE)
+@ApiModel("CustomSecretManager")
+@OwnedBy(HarnessTeam.PL)
+@Schema(name = "CustomSecretManager", description = "This contains details of Custom Secret Manager connectors")
+public class CustomSecretManagerConnectorDTO
+    extends ConnectorConfigDTO implements DelegateSelectable, ManagerExecutable {
```
Comment (meenakshi-harness) [inline 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/customsecretmanager/CustomSecretManagerConnectorDTO.java:49]: Remove ManagerExecutable - This is for supporting connector validation via manager.

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/customsecretmanager/CustomSecretManagerConnectorDTO.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  @Schema @JsonIgnore private boolean harnessManaged;
+
+  @SecretReference
+  @ApiModelProperty(dataType = "string")
+  @Schema(description = SecretManagerDescriptionConstants.CUSTOM_AUTH_TOKEN)
+  private SecretRefData connectorRef;
+
+  private String host;
+  private String workingDirectory;
+  private TemplateLinkConfig template;
```
Comment (meenakshi-harness) [inline 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/customsecretmanager/CustomSecretManagerConnectorDTO.java:62]: not null

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/customsecretmanager/TemplateLinkConfig.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+@Data
+@NoArgsConstructor
+@JsonIgnoreProperties(ignoreUnknown = true)
+@FieldDefaults(level = AccessLevel.PRIVATE)
+@TypeAlias("delegate.beans.connector.customsecretmanager.TemplateLinkConfig")
+@OwnedBy(PL)
+public class TemplateLinkConfig {
+  @NotNull String templateRef;
+  @Pattern(regexp = NGRegexValidatorConstants.VERSION_LABEL_PATTERN) String versionLabel;
```
Comment (meenakshi-harness) [inline 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/customsecretmanager/TemplateLinkConfig.java:33]: This is also NotNull

**Review by meenakshi-harness — CHANGES_REQUESTED** [review]

**Pull request closed without merging**